domains/environment_domains/dns.tf

```diff
@@ -3,5 +3,47 @@ data "azurerm_dns_zone" "main" {
   resource_group_name = var.resource_group_name
 }
 
-# TODO: Create validation TXT record
-# TODO: Create domain ALIAS A record
+resource "azurerm_dns_txt_record" "main" {
+  for_each            = { for k in toset(var.domains) : k => k if k != "apex" }
+  name                = join(".", ["_dnsauth", "${each.key}"])
+  zone_name           = data.azurerm_dns_zone.main.name
+  resource_group_name = var.resource_group_name
+  ttl                 = 3600
+
+  record {
+    value = azurerm_cdn_frontdoor_custom_domain.main[each.key].validation_token
+  }
+}
+
+resource "azurerm_dns_txt_record" "apex" {
+  for_each            = { for k in toset(var.domains) : k => k if k == "apex" }
+  name                = "_dnsauth"
+  zone_name           = data.azurerm_dns_zone.main.name
+  resource_group_name = var.resource_group_name
+  ttl                 = 3600
+
+  record {
+    value = azurerm_cdn_frontdoor_custom_domain.main[each.key].validation_token
+  }
+}
+
+resource "azurerm_dns_cname_record" "main" {
+  depends_on = [azurerm_cdn_frontdoor_route.main]
+  for_each   = { for k in toset(var.domains) : k => k if k != "apex" }
+
+  name                = each.key
+  zone_name           = data.azurerm_dns_zone.main.name
+  resource_group_name = var.resource_group_name
+  ttl                 = 3600
```

Comment:
3600 is fine for validation domains. Can we set it to 300 for actual records

Reply:
Updated to 300